modules/nf-core/fastqscreen/buildfromindex/main.nf

```diff
@@ -0,0 +1,67 @@
+process FASTQSCREEN_BUILDFROMINDEX {
+    label 'process_single'
+
+    conda "fastq-screen"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fastq-screen%3A0.15.3--pl5321hdfd78af_0':
+        'quay.io/biocontainers/fastq-screen:0.15.3--pl5321hdfd78af_0'}"
+
+    input:
+    val(id)
+    path(index), stageAs: "dir*"
+
+    output:
+    path("FastQ_Screen_Genomes"), emit: database
+    path "versions.yml"         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    dir = "FastQ_Screen_Genomes"
+    folder = index.collect { it.toString() }
+    database = [id, folder].transpose()
+    copy_index = folder.collect { "cp -r ${it} $dir/${it}"}.join(" && ")
+
+    // Folder name and index (within folder) name could be different - use bash to look for index name
+    config = database
+        .collect { "########## ${it[0]} \nDATABASE ${it[0]} $dir/${it[1]}/${it[1] + '_to_be_replaced'}" }
+        .join("\n\n")
+
+    """
+    mkdir $dir
+    $copy_index
+
+    echo "$config"  >> $dir/fastq_screen.conf
+    echo "Replace folder name real index name"
+
+    for f in ${folder.join(' ')}
+    do
+        TO_REPLACE=\${f}_to_be_replaced
+
+        REPLACE_WITH=(\$f"/*.rev.1.bt2")
+        REPLACE_WITH=\$(basename \$REPLACE_WITH)
+        REPLACE_WITH=\${REPLACE_WITH%%.rev.1.bt2}
+
+        sed "s/\$TO_REPLACE/\$REPLACE_WITH/g" $dir/fastq_screen.conf > new_conf.conf
+        mv new_conf.conf $dir/fastq_screen.conf
+    done
+
+    cat <<-END_VERSIONS > versions.yml
+
+    "${task.process}":
+        fastqscreen: \$(echo \$(fastq_screen --version 2>&1) | sed 's/^.*FastQ Screen v//; s/ .*\$//')
+    END_VERSIONS
+    """
+
+    stub:
+    """
+    mkdir $dir
+    touch $dir/fastq_screen.conf
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        fastqscreen: \$(echo \$(fastq_screen --version 2>&1) | sed 's/^.*FastQ Screen v//; s/ .*\$//')
+    END_VERSIONS
```

Comment:
```suggestion
    config = database
        .collect { "########## ${it[0]} \nDATABASE ${it[0]} $dir/${it[1]}/${it[1] + '_to_be_replaced'}" }
        .join("\n\n")
        .replace("\n", "\\n")

    """
    mkdir $dir
    $copy_index

    echo "$config" >> $dir/fastq_screen.conf
    sed -i 's/\\\\n/\\n/g' $dir/fastq_screen.conf
    echo "Replace folder name real index name"

    for f in ${folder.join(' ')}
    do
        TO_REPLACE=\${f}_to_be_replaced

        REPLACE_WITH=(\$f"/*.rev.1.bt2")
        REPLACE_WITH=\$(basename \$REPLACE_WITH)
        REPLACE_WITH=\${REPLACE_WITH%%.rev.1.bt2}

        sed "s/\$TO_REPLACE/\$REPLACE_WITH/g" $dir/fastq_screen.conf > new_conf.conf
        mv new_conf.conf $dir/fastq_screen.conf
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastqscreen: \$(echo \$(fastq_screen --version 2>&1) | sed 's/^.*FastQ Screen v//; s/ .*\$//')
    END_VERSIONS
    """

    stub:
    """
    mkdir $dir
    touch $dir/fastq_screen.conf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastqscreen: \$(echo \$(fastq_screen --version 2>&1) | sed 's/^.*FastQ Screen v//; s/ .*\$//')
    END_VERSIONS
```
Picking up on the discussion that was started in https://github.com/nf-core/modules/issues/3157, the issue with the tests seems to be that the new line characters in `config` confuse nextflow and break the end of the script.

Escaping the newline characters and substituting them later seems to solve this issue. It's not very pretty but at least it solves the problem.

Reply:
If it works, then let's go for it. The previous code was not pretty either.